sklearn/utils/tests/test_stats.py

```diff
@@ -48,23 +66,35 @@ def test_weighted_percentile_zero_weight_zero_percentile():
     assert approx(score) == 4
 
 
-def test_weighted_median_equal_weights():
+@pytest.mark.parametrize(
+    "array_namespace, device, dtype_name", yield_namespace_device_dtype_combinations()
+)
+def test_weighted_median_equal_weights(array_namespace, device, dtype_name):
+    xp = _array_api_for_tests(array_namespace, device)
     # Checks weighted percentile=0.5 is same as median when weights equal
     rng = np.random.RandomState(0)
     # Odd size as _weighted_percentile takes lower weighted percentile
     x = rng.randint(10, size=11)
-    weights = np.ones(x.shape)
+    x = x.astype(dtype_name, copy=False)
+    x = xp.asarray(x, device=device)
```

Comment:
What I have learned talking to maintainers is that in case we have our data on a specific device, we should leave it there and avoid moving it to another device, because this is costly.
In this case, when we have a numpy array from the beginning with the `np.random.RandomState(0)`, this might mean maybe **not** to parametrise this test with Array API? Or maybe we would not use a numpy random seed? 

I'm unsure here. What do others think?
This also applies to the other tests below.

Reply:
I think in tests it is fine to move data around. For example when we load a dataset or generate random input data, the only chance we have to test how it would work on a GPU is to move the data there first.

As a side note, I am starting to wonder how big of a problem "moving data around" really is. I think we should generally avoid it, just like making unnecessary copies, but sometimes I think life is easier if we move things. For example we have an estimator (somewhere in preprocesing I think?) where we decided that moving `y` to the CPU memory and using Numpy is probably fast (how big is a typical `y`? Millions of floats?) and much less complicated in terms of implementation.